frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkLoadManager.java

```diff
@@ -112,7 +112,19 @@ public int getNumRunningWorkers(String modelName) {
                 for (int i = currentWorkers - 1; i >= maxWorker; --i) {
                     // TODO: kill unhealthy worker first.
                     WorkerThread thread = threads.remove(i);
+                    WorkerLifeCycle lifecycle = thread.getLifeCycle();
                     thread.shutdown();
+
+                    Process workerProcess = lifecycle.getProcess();
+
+                    if (workerProcess.isAlive()) {
+                        workerProcess.destroyForcibly();
+                        try {
+                            workerProcess.waitFor();
+                        } catch (InterruptedException e) {
```

Comment:
What could cause the process wait to be interrupted? Asking just out of curiosity. 

It seems like the future is not completed. All the other returns are returning a completed future. This might cause some issues. 

This method is called in UnregisterModel (DELETE /models) as well as scale up/down models (PUT /models). You should test for both. Could you test for both to see if anything breaks by returning just the future.

Reply:
This only happens in MMS' context if another thread explcitly calls waitingThread.interrupt before or during the waitFor which never happens. I think it's probably better to complete the future w/ an error before returning here though.

Reply:
I think you added future to this, so I will resolve this comment.